docs/services/azure-services-guide.md

```diff
@@ -0,0 +1,625 @@
+```
+
+3. Enable SQL Service Endpoint
+
+Get name of AKS node Virtual Network:
+```bash
+export AKS_VNET=$(az network vnet list -g $AKS_RESOURCE_GROUP | jq -j '.[0].name')
+```
+
+Add Service Endpoint to AKS subnet:
+```bash
+az network vnet subnet update -g $AKS_RESOURCE_GROUP --vnet-name $AKS_VNET -n aks-subnet --service-endpoints Microsoft.Sql
+```
+
+4. Connect to AKS Cluster 
+```bash
+az aks get-credentials --resource-group $RESOURCE_GROUP_NAME --name $AKS_NAME
+```
+
+5. Make sure `kubectl` is able to communicate with AKS Cluster
+```bash
+kubectl cluster-info
+```
+
+### Crossplane
+Using the newly provisioned cluster:
+
+* Install Crossplane from alpha channel using the [Crossplane Installation
+  Guide](../install-crossplane.md#alpha)
+* Install the Azure stack into Crossplane using the [Azure stack
+  section](../install-crossplane.md#azure-stack) of the install guide.
+* Obtain [Cloud Provider Credentials](../cloud-providers.md)
+
+#### Infrastructure Namespaces
+Kubernetes namespaces allow for separation of environments within your cluster.
+You may choose to use namespaces to group resources by team, application, or any
+other logical distinction. For this demo, we will create a namespace called
+`app-project1-dev`, which we will use to group our Azure infrastructure
+components.
+
+* Define a `Namespace` in `azure-infra-dev.yaml`:
+```yaml
+cat > azure-infra-dev.yaml <<EOF
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: azure-infra-dev
+EOF
+```
+
+* Create the `Namespace`:
+```bash
+kubectl create -f azure-infra-dev.yaml.yaml
+```
+
+* You should see the following output:
+```bash
+namespace/azure-infra-dev.yaml created
+```
+
+#### Azure Provider
+It is essential to make sure that the Azure Service Principal is configured with
+all permissions outlined in the [provider
+guide](../cloud-providers/azure/azure-provider.md).
+
+Using Azure Service Principal `crossplane-azure-provider-key.json`:
+* Generate BASE64ENCODED_AZURE_PROVIDER_CREDS encoded value:
+```bash
+export BASE64ENCODED_AZURE_PROVIDER_CREDS=$(base64 crossplane-azure-provider-key.json | tr -d "\n")
+```
+
+* Define an Azure `Provider` and `Secret` in `azure-provider.yaml`:
+```yaml
+cat > azure-provider.yaml <<EOF
+---
+# Azure Admin service account secret - used by Azure Provider
+apiVersion: v1
+kind: Secret
+metadata:
+  name: demo-provider-azure-dev
+  namespace: azure-infra-dev
+type: Opaque
+data:
+  credentials: BASE64ENCODED_AZURE_PROVIDER_CREDS
+---
+# Azure Provider with service account secret reference - used to provision resources
+apiVersion: azure.crossplane.io/v1alpha2
+kind: Provider
+metadata:
+  name: demo-azure
+  namespace: azure-infra-dev
+spec:
+  credentialsSecretRef:
+    name: demo-provider-azure-dev
+    key: credentials
+EOF
+```
+
+#### Create Provider
+* Create Azure provider:
+```bash
+sed "s/BASE64ENCODED_AZURE_PROVIDER_CREDS/$BASE64ENCODED_AZURE_PROVIDER_CREDS/g" azure-provider.yaml | kubectl create -f -
```

Comment:
In the stacks guide and the GCP stuff that Muvaffak and I have been working on, we've been embedding this in the heredoc that is `cat`ted instead of using `sed`. Both approaches work fine, but we should standardize on one. I would suggest the `cat` approach since it seems like fewer steps. Here's a [link to an example](https://github.com/crossplaneio/crossplane/pull/803/files#diff-13b58326c3314fa933bd255157c8fa59R146)

Reply:
I like the pattern that is used in your stacks doc and will apply here :)